src/rust/ensogl/src/display/scene.rs

```diff
@@ -120,23 +120,63 @@ pub enum Target {
 }
 
 impl Target {
+
+    /// Decode the symbol_id and instance_id that was encoded in the `fragment_runner`.
+    ///
+    /// See the `encode` method for more information on the encoding.
+    fn decode(pack1:u32, pack2:u32, pack3:u32) -> (u32,u32) {
+        let value1 = (pack1 << 4) + (pack2 >> 4);
+        let value2 = pack3 + ((pack2 & 0x000F) << 8);
+        (value1, value2)
+    }
+
+    /// Encode two u32 values into three u8 values.
+    /// This is the same encoding that is used in the `fragment_runner`.
+    /// This encoding is lossy.
+    ///
+    /// We use 12 bits from each value and pack them into the 3 output bites like
+    /// described in the following diagram.
+    ///
+    ///  Input
+    ///
+    ///  value1 as bytes                    value2 as bytes
+    /// -------------------------           -------------------------
+    /// | v1a | v1b | v1c | v1d |           | v2a | v2b | v2c | v2d |
+    /// -------------------------           -------------------------
+    ///
+    ///  Output
+    ///
+    /// byte1                      byte2                      byte3
+    /// -----------------------    -----------------------     -------
+    /// | v2c[4..8] v1d[0..4] |    | v1d[4..8] v2c[4..8] |     | v2d |
```

Comment:
The arrays do not show how bounds behave, please use Rust (`[x..=y]`) notion to show that. Plus, are you sure the drawings are correct? I cannot get my head around how they work (there is for example `v2c[4..8]` two times and they are strangely layouted).